compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1955,10 +1984,23 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
                     }
                 }
                 identifiers.put(identifier, symbol);
+                newIdentifiers.put(identifier, symbol);
```

Comment:
Can't we use the existing identifiers list instead of creating a new one?

Reply:
To identify the identifiers defined in the current clause we use this.